vtex-apps/structured-data
# Use correct product URL as schema id

**What problem is this solving?**


The current schema generated for products uses the full URL - using `portal.vtexcommercestable.com.br` - as its `@id`. In order to provide a better path to search engines and avoid that URL to be indexed, this PR builds the correct URL to the product and use it as the schema `id`.

**How should this be manually tested?**

[This workspace](https://fixidschema--arcaplanet.myvtex.com/acana-pacifica-per-cani/p) has the branch linked, and inspecting the HTML, we can see the following schema:

```json
{
  "@context": "https://schema.org/",
  "@type": "Product",
  "@id": "https://fixidschema--arcaplanet.myvtex.com/acana-pacifica-per-cani/p",
  "name": "Acana Dog Pacifica",
  "brand": "ACANA",
  "image": "https://arcaplanet.vtexassets.com/arquivos/ids/234489/acana-cane-pacifica.jpg?v=637454872023770000",
  "description": "Acana Dog Pacifica, secco per cani adulti di tutte le taglie ed età, con pesce fresco come salmone e aringa canadesi per una dieta bilanciata. Compralo ora!",
  "mpn": "8498",
  "sku": "7770",
  "category": "Cibo Secco",
  "offers": {
    // omitted for brevity
  }
}
```

To compare, [here](https://www.arcaplanet.it/acana-pacifica-per-cani/p) we have the same schema running on production.

```json
{
  "@context": "https://schema.org/",
  "@type": "Product",
  "@id": "https://portal.vtexcommercestable.com.br/acana-pacifica-per-cani/p",
  "name": "Acana Dog Pacifica",
  "brand": "ACANA",
  "image": "https://arcaplanet.vtexassets.com/arquivos/ids/234489/acana-cane-pacifica.jpg?v=637454872023770000",
  "description": "Acana Dog Pacifica, secco per cani adulti di tutte le taglie ed età, con pesce fresco come salmone e aringa canadesi per una dieta bilanciata. Compralo ora!",
  "mpn": "8498",
  "sku": "7770",
  "category": "Cibo Secco",
  "offers": {
    // omitted for brevity
  }
}
```
